File: backend/src/main/java/es/ual/dra/agrotrack/dto/response/CultivoResponse.java
package es.ual.dra.agrotrack.dto.response;

import es.ual.dra.agrotrack.model.entity.CultivoParcela;
import es.ual.dra.agrotrack.model.enums.EstadoCultivo;

import java.time.LocalDate;

public record CultivoResponse(
    Long id,
    Long parcelaId,
    String parcelaNombre,
    Long productoId,
    String productoNombre,
    LocalDate fechaSiembra,
    EstadoCultivo estado,
    String notas
) {
    public static CultivoResponse from(CultivoParcela c) {
        return new CultivoResponse(
            c.getId(),
            c.getParcela().getId(),
            c.getParcela().getNombre(),
            c.getProducto().getId(),
            c.getProducto().getNombre(),
            c.getFechaSiembra(),
            c.getEstado(),
            c.getNotas()
        );
    }
}

File: frontend/src/app/models/response/cultivo-response.ts
import { EstadoCultivo } from '../enums/estado-cultivo';

export interface CultivoResponse {
  id: number;
  parcelaId: number;
  parcelaNombre: string;
  productoId: number;
  productoNombre: string;
  fechaSiembra: string;
  estado: EstadoCultivo;
  notas: string | null;
}
